lib/Runtime/Language/SourceTextModuleRecord.cpp

```diff
@@ -547,10 +555,36 @@ namespace Js
         return !ambiguousResolution;
     }
 
+    void SourceTextModuleRecord::SetParent(SourceTextModuleRecord* parentRecord, LPCOLESTR moduleName)
+    {
+        Assert(parentRecord != nullptr);
+        Assert(parentRecord->childrenModuleSet != nullptr);
+        if (!parentRecord->childrenModuleSet->ContainsKey(moduleName))
+        {
+            parentRecord->childrenModuleSet->AddNew(moduleName, this);
+
+            if (this->parentModuleList == nullptr)
+            {
+                Recycler* recycler = GetScriptContext()->GetRecycler();
+                this->parentModuleList = RecyclerNew(recycler, ModuleRecordList, recycler);
+            }
+            bool contains = this->parentModuleList->Contains(parentRecord);
```

Comment:
Nitpick: We don't need to see if parentRecord is in the list if we just constructed the list.